diff -ur a/src/XMonad/Main.hs b/src/XMonad/Main.hs
--- a/src/XMonad/Main.hs	2017-02-07 20:28:53.000000000 +0000
+++ b/src/XMonad/Main.hs	2017-10-24 17:04:33.196180072 +0100
@@ -368,7 +368,9 @@
         dpy <- asks display
         root <- asks theRoot
         (_, _, w', _, _, _, _, _) <- io $ queryPointer dpy root
-        when (w == w') (focus w)
+        -- when Xlib cannot find a child that contains the pointer,
+        -- it returns None(0)
+        when (w' == 0 || w == w') (focus w)
 
 -- left a window, check if we need to focus root
 handle e@(CrossingEvent {ev_event_type = t})
